feature/BL-41-create-all-other-crud-api

vbado opened the pull request.

vbado committed:
[BL-41] add not null for FK

vbado committed:
[BL-41] add Measurement

vbado committed:
[BL-41] finish the tests for Read, Update, Delete, and List operations

vbado committed:
[BL-41] add Measurement test and other

vacoola1 commented on src/main/resources/db/changelog/biolab.changelog.yml:
```diff
@@ -2,4 +2,6 @@ databaseChangeLog:
\ No newline at end of file
+      file: db/changelog/changeset/1_bl_28.yml
+  - include:
+      file: db/changelog/changeset/1_bl_41.yml
```
I think we can still update 1_bl_21.yml

              - column:
                  name: measurement_id
                  type: integer
                  constraints:
                    nullable: false

vbado replied:
you said thet you changed in file, but it as not correct. so i created a file version. edit old file - easy

vacoola1 replied:
its not correct when you already have databases to break.. But currently we dont have. So i wanna make it easy to read

vacoola1 commented on src/test/resources/application-test.yml:
```diff
@@ -1,16 +1,16 @@
-  liquibase:
-    change-log: db/changelog/biolab.changelog.yml
\ No newline at end of file
+#spring:
```
Q: Do tests work without this?

vbado replied:
yes

vacoola1 commented on src/main/java/com/bmlab/launchpad/service/MeasurementService.java:
```diff
@@ -0,0 +1,75 @@
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class MeasurementService {
```
I made another PR, BL-47, to improve services. Pls review and merge to your branch first

vacoola1 commented on src/test/java/com/bmlab/launchpad/web/controller/MetricControllerIntegrationTest.java:
```diff
@@ -63,7 +63,7 @@ void tearDown() {
     @DisplayName("Create")
     class CreateTests {
-        @Disabled
+        //@Disabled
```
just delete it

vbado committed:
[BL-41] Merge branch 'develop' into feature/BL-41-create-all-other-crud-api

# Conflicts:
#	src/test/java/com/bmlab/launchpad/web/controller/MetricControllerIntegrationTest.java

vbado committed:
[BL-41] edit witch generic

vbado committed:
[BL-41] fix return 501

vbado committed:
[BL-41] dell 1_bl_41.yml and move to 1_bl_21.yml
